ios/FluentUI/Extensions/UIColor+Extensions.swift

```diff
@@ -5,35 +5,44 @@
-    var current: UIColor {
-        return self
+    /// Creates a UIColor from a single 32-bit integer value.
+    ///
+    /// - Parameter hexValue: Integer value, generally represented as hexadecimal (e.g. `0x0086F0`), to use to initialize this color.
+    ///     Must be formatted as ARGB. Note: we will not utilize the alpha channel.
+    convenience init(hexValue: UInt32) {
+        self.init(
+            red: CGFloat((hexValue & 0xFF0000) >> 16) / 255.0,
+            green: CGFloat((hexValue & 0x00FF00) >> 8) / 255.0,
+            blue: CGFloat((hexValue & 0x0000FF) >> 0) / 255.0,
+            alpha: 1.0)
     }
 
-    private var light: UIColor? {
-        get {
-            return objc_getAssociatedObject(self, &AssociatedKeys.light) as? UIColor
-        }
-        set {
-            objc_setAssociatedObject(self, &AssociatedKeys.light, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
-        }
-    }
-    private var lightHighContrast: UIColor? {
-        get {
-            return objc_getAssociatedObject(self, &AssociatedKeys.lightHighContrast) as? UIColor
-        }
-        set {
-            objc_setAssociatedObject(self, &AssociatedKeys.lightHighContrast, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
-        }
-    }
-
-    convenience init(light: UIColor, lightHighContrast: UIColor? = nil, lightElevated: UIColor? = nil, lightElevatedHighContrast: UIColor? = nil, dark: UIColor? = nil, darkHighContrast: UIColor? = nil, darkElevated: UIColor? = nil, darkElevatedHighContrast: UIColor? = nil) {
+    /// Creates a dynamic color object that returns the appropriate color value based on the current
+    /// rendering context.
+    ///
+    /// The decision order for choosing between the colors is based on the following questions, in order:
+    /// - Is the current `userInterfaceStyle` `.dark` or `.light`?
+    /// - Is the current `userInterfaceLevel` `.base` or `.elevated`?
+    /// - Is the current `accessibilityContrast` `.normal` or `.high`?
+    ///
+    /// - Parameter light: The default color for a light context. Required.
+    /// - Parameter lightHighContrast: The override color for a light, high contrast context. Optional.
+    /// - Parameter lightElevated: The override color for a light, elevated context. Optional.
+    /// - Parameter lightElevatedHighContrast: The override color for a light, elevated, high contrast context. Optional.
+    /// - Parameter dark: The override color for a dark context. Optional.
+    /// - Parameter darkHighContrast: The override color for a dark, high contrast context. Optional.
+    /// - Parameter darkElevated: The override color for a dark, elevated context. Optional.
+    /// - Parameter darkElevatedHighContrast: The override color for a dark, elevated, high contrast context. Optional.
```

Comment:
should we say increased contrast?

Reply:
High contrast is the term that fluent is using. I'll see if we can get it adjusted, but that'll also make this a much bigger diff so I'll research now and possibly address later.